zebra-state/src/request.rs

```diff
@@ -607,6 +607,12 @@ pub enum Request {
     /// * [`Response::BlockHash(None)`](Response::BlockHash) otherwise.
     BestChainBlockHash(block::Height),
 
+    /// Checks if a block is present anywhere in the state service; Looks up hash in block stores.
+    ///
+    /// Returns [`Response::BlockLocation(Some(Location))`](Response::BlockLocation) if the block is in the best state service.
+    /// Returns [`Response::BlockLocation(None)`](Response::BlockLocation) otherwise.
+    Contains(block::Hash),
```

Comment:
Can we rename this request so it's clear that the block is known, but might not be valid?
(Also it would be helpful to have "block" in the name!)

```suggestion
    KnownBlock(block::Hash),
```